src/Compilers/Core/Portable/PEWriter/PdbWriter.cs

```diff
@@ -559,14 +558,17 @@ private static Type GetCorSymWriterSxSType()
 
         public void SetMetadataEmitter(MetadataWriter metadataWriter)
         {
-            Stream streamOpt = _streamProvider();
+            Stream stream = _streamProvider() ?? new System.IO.MemoryStream();
 
             try
             {
                 var instance = (ISymUnmanagedWriter2)(_symWriterFactory != null ? _symWriterFactory() : Activator.CreateInstance(GetCorSymWriterSxSType()));
-                var comStream = (streamOpt != null) ? new ComStreamWrapper(streamOpt) : null;
 
-                instance.Initialize(new PdbMetadataWrapper(metadataWriter), _fileName, comStream, fullBuild: true);
+                // Important: If the stream is not specified or if it is non-empty the SymWriter appends data to it (provided it contains valid PDB)
+                // and the resulting PDB has Age = existing_age + 1.
```

Comment:
Actually, we might want to do that to generated deterministic PDB... work in progress.